src/overlays/actors/ovl_En_Toryo/z_en_toryo.c

```diff
@@ -281,64 +281,64 @@ s32 func_80B206A0(EnToryo* this, PlayState* play) {
     return ret;
 }
 
-void func_80B20768(EnToryo* this, PlayState* play) {
+void EnToryo_Talk(EnToryo* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
-    s16 sp32;
-    s16 sp30;
+    s16 posX;
+    s16 posY;
 
-    if (this->unk_1E4 == 3) {
+    if (this->messageState == 3) {
         Actor_ProcessTalkRequest(&this->actor, play);
         Message_ContinueTextbox(play, this->actor.textId);
-        this->unk_1E4 = 1;
+        this->messageState = 1;
     }
 
-    if (this->unk_1E4 == 1) {
-        this->unk_1E4 = func_80B203D8(this, play);
+    if (this->messageState == 1) {
+        this->messageState = EnToryo_TalkRespond(this, play);
     }
 
-    if (this->unk_1E4 == 5) {
-        this->unk_1E4 = func_80B205CC(this, play);
+    if (this->messageState == 5) {
+        this->messageState = EnToryo_DoneTalking(this, play);
         return;
     }
 
-    if (this->unk_1E4 == 2) {
+    if (this->messageState == 2) {
         Message_ContinueTextbox(play, this->actor.textId);
-        this->unk_1E4 = 1;
+        this->messageState = 1;
     }
 
-    if (this->unk_1E4 == 4) {
+    if (this->messageState == 4) {
         if (Actor_HasParent(&this->actor, play)) {
             this->actor.parent = NULL;
-            this->unk_1E4 = 5;
+            this->messageState = 5;
         } else {
             func_8002F434(&this->actor, play, GI_BROKEN_GORONS_SWORD, 100.0f, 10.0f);
         }
         return;
     }
 
-    if (this->unk_1E4 == 0) {
+    if (this->messageState == 0) {
         if (Actor_ProcessTalkRequest(&this->actor, play)) {
-            this->unk_1E0 = func_8002F368(play);
-            if (this->unk_1E0 != 0) {
-                player->actor.textId = func_80B20634(this, play);
+            this->exchangeItemId = func_8002F368(play);
+            if (this->exchangeItemId != EXCH_ITEM_NONE) {
+                player->actor.textId = EnToryo_OfferItem(this, play);
                 this->actor.textId = player->actor.textId;
             }
-            this->unk_1E4 = 1;
+            this->messageState = 1;
             return;
         }
 
-        Actor_GetScreenPos(play, &this->actor, &sp32, &sp30);
-        if ((sp32 >= 0) && (sp32 < 0x141) && (sp30 >= 0) && (sp30 < 0xF1)) {
-            this->actor.textId = func_80B206A0(this, play);
+        Actor_GetScreenPos(play, &this->actor, &posX, &posY);
+        if ((posX >= 0) && (posX <= SCREEN_WIDTH) && (posY >= 0) && (posY <= SCREEN_HEIGHT)) {
+            this->actor.textId = EnToryo_GetTextId(this, play);
             func_8002F298(&this->actor, play, 100.0f, EXCH_ITEM_POACHERS_SAW);
         }
     }
 }
 
-void func_80B20914(EnToryo* this, PlayState* play) {
+void EnToryo_TalkAction(EnToryo* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
```

Comment:
usually we like to name the action functions as just the verb itself (Tall/Walk/Run/Whatever)
The function that you called _Talk id say is more of a lower level routine like `UpdateMessageState`/`HandleDialogue` or something similar (not set on those names, just brainstorming)

This action runs, really, during the duration of this actors lifetime. So its more than just "Talk", which makes it actually pretty hard to name. Its handles both idle and talking. So I dont have a sufficient suggestion yet

Reply:
I can understand why calling this "Action" since it's the only function `actionFunc` is set to 8)

I don't have outstanding ideas either
`EnToryo_Main`
`EnToryo_UpdateTalk`
`EnToryo_Action`
`EnToryo_Be`
`EnToryo_BeThere`
`EnToryo_Live`
`EnToryo_Act`
... :shrug:

Reply:
Personally, I think `EnToryo_Idle` would be good enough.

Reply:
think i agree, mainly because if you look at it from the perspective of what "action" it handles most, the actor is going to be idle 99% of the time.

if the actor was implemented "properly" this function would have then passed execution on to a proper talking action, instead of just calling a function directly on its own.
But yeah idk, annoying situation all around

Reply:
So is anyone expecting `_Main` to be changed to `_Idle` ? (I'd be fine with it personally)

Reply:
yes, id like it changed